### DISCUSSION TITLE
bluetooth paired, audio still plays out the RPI.

### DISCUSSION THREAD

**Discussion by subatomicglue** [rern/rAudio#600] [Fixed Issues] [answered]
**UPDATE:** scroll to end for the audio/bluetooth issue https://github.com/rern/rAudio-1/discussions/600#discussioncomment-1616632   (I ended up solving the first issue below)
----------------------------------------------------------------------
original topic was: Database not available with USB FAT32 stick, rpi 4b, using rAudio-1-64bit-20211022 img
-------
Hi, I have a usb stick with m4a and mp3 files in subfolders inserted into my `raspberry pi 4b`.

I used belena etcher with `rAudio-1-64bit-20211022.img.xz` on a 16gb card
Wifi is connected.   
USB stick is 256 GB with a few deeply nested folders totaling ~5gb in m4a/mp3 files. 
MP3 and M4A files are located maybe 3-4 levels down the folder hierarchy.
Under System, my USB stick shows up, named with the right label that I gave it.  (/dev/sda1  4.7GB / 250GB)
I can access the device using my phone (from the QR code), same menus though.
`Settings / Player` seems to never display, just shows the pulsating blue square in the middle of the screen.
I've done nothing else.

When I tap on `Library / USB`  (I'm trying to play some music from the USB stick)

```
Library Database
------
Database not available for this location
To populate new files to database:
Settings -> Library (||\)
          [ok]
```

When I go to the `Settings / Library`, there is only a bunch of checkboxes on 2 pages of options.   
No option to "rescan" or "rebuild" the library.
I've been through all the locations under `Settings`, and still dont see any way to "rescan" or "rebuild" the library.

Maybe if there is no library, the software could default to Folder browsing mode?  Or...  prompt me to "rescan" the library perhaps?

Honestly, I'd love to simply play folders/files directly off the USB stick and forego any special pre-scanned "library".
My stick is organized!   (i'm one of those users in Plex who uses the "Folder" view, ha)

Ideas?

**Comment by subatomicglue** [comment]
# Solved:
I found how to scan - 

1.) the display is very hard to read because everything is very small on my 7" touchscreen
2.) there appears to be 2 menu items on the same line under `Settings / Library`, a little button off to the right.   I didn't realize that could be selected.   Once I tapped that, the library built.     

Probably want to make that easier to find.  Or multiple ways to find it - like the box that warns you there is no database could allow you to trigger a rescan;  and also maybe add a more obvious "rebuild" button under `settings / system` in addition to the convenient (but hard to find) one under `Settings / Library`...   Why not make it findable :)   kind of a core feature.

Additionally, I noticed that the `Settings / Player` screen works now (doesn't hang anymore).   I guess when the database isn't present, it doesn't know what to do.

# New Problem:
Looks like m4a files aren't supported.... bummer.

**Comment by subatomicglue** [comment]
# Solved:
Ok, I found that m4a files is disabled by default, and to get it back, I went inside `Settings / Player / Options / FFMPEG (set to on)`
Rescan, and all my m4a files play now.

# New Problem:
Now, bluetooth isn't working.   It PAIRS ok with my JAMBOX, and it says bluetooth is the output under `Settings / Player / Output / Bluetooth` 

but audio keeps coming out my pi...    even though the JAMBOX is paired...   weird... 

Inside `Settings / Player / Output`, The original audio output selector dropdown for HDMI vs Headphones is now gone, and there is only Bluetooth when my bluetooth device is paired....  Not that it is making any sound.   The JAMBOX even tells me pairing was successful, and the rAudio-1 software under `Settings / Networks / Bluetooth` is telling me my JAMBOX is paired with a little green light there, and I can forget it if I want to, and the JAMBOX makes a sound when I do that.

I did reboot, just in case that was needed :)

(I got a 7" touchscreen with built in speakers, so I play audio out the HDMI normally.)

Ideas?

**Reply by phreatic62** [reply]
I have a similar problem, see my comment in #595.

**Comment by subatomicglue** [comment]
Split between two settings screens,  NETWORKS and PLAYER....  it's not like player allows you to select anything but the bluetooth output.   But, dont be fooled by this menu, audio is still coming out the RPI HDMI output, and not from the bluetooth speaker :) :) :) :)

![IMG_3418](https://user-images.githubusercontent.com/7660853/141405722-01b2dd4d-d813-4aff-b8ab-409c3cfab61b.jpg)
![IMG_3419](https://user-images.githubusercontent.com/7660853/141405730-e3f3d380-f23b-4dfd-b906-a069c814619a.jpg)

**Comment by subatomicglue** [comment]
When I turn off my bluetooth speaker - the audio coming out the rAudio-1  (RPI 4b) pauses for a good 10 seconds, then resumes.   It's aware of the bluetooth speaker for sure.

**Comment by phreatic62** [comment]
I've just done the latest update, 2021112 and had to re-pair, but all seems to work now.
I may do a backup of the card I have and try a fresh install to see if there is something in my old backup that does not get updated yet is in the new distribution.

**Reply by subatomicglue** [reply]
I hit the "update" last night before posting the above...   I've never heard audio out the Bluetooth yet

Using 20211022 with "update" applied as of last night

```
but all seems to work now.
```
when you say "all" do you mean you can hear audio on your bluetooth speaker?   Is there some trick or setting I need to enable?

**Reply by subatomicglue** [reply]
Solved: 20211112 gets bluetooth working for me now

**Comment by subatomicglue** [comment, marked as answer]
Just tried the new 20211112 release, and BLUETOOTH WORKS.    All I did was switch on my speaker and it auto paired, and transferred audio over to play on it

THANK YOU rAudio-1 team!
